examples/virtual_methods/base_button/imp.rs

```diff
@@ -0,0 +1,103 @@
+use super::BaseButtonExt;
+use gtk::{
+    gio,
+    glib::{self, Error},
+    prelude::*,
+    subclass::prelude::*,
+};
+use std::{future::Future, pin::Pin};
+
+pub type BaseButtonInstance = super::BaseButton;
+pub type PinnedFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + 'static>>;
+
+/// GObject glue code for our BaseButtonClass which holds the function pointers to our virtual functions.
+#[repr(C)]
+pub struct BaseButtonClass {
+    pub parent_class: gtk::ffi::GtkButtonClass,
+    pub sync_method: Option<unsafe fn(&BaseButtonInstance, extra_text: Option<String>)>,
+    pub async_method: Option<unsafe fn(&BaseButtonInstance) -> PinnedFuture>,
```

Comment:
Maybe add a comment here that `extern "C"` functions and FFI-safe types would be necessary if this should be called from C or other languages